docs/docs/creating-and-modifying-pages.md

```diff
@@ -160,6 +160,112 @@ exports.onCreatePage = async ({ page, boundActionCreators }) => {
 };
 ```
 
+### Client Route Params
+
+Gatsby can create client-only routes which take paramaters. 
+
+We'll walk quickly through how to setup a route on your site like `/widgets/view/ID`.
+
+**Build config:**
+
+First configure `gatsby-node.js`:
+
+```javascript
+exports.onCreatePage = async ({ page, boundActionCreators }) => {
+  const { createPage } = boundActionCreators;
+
+  return new Promise((resolve, reject) => {
+    // Add client route for all paths matching `/view/*`
+    if (/view/.test(page.path)) {
+      // Gatsby paths have a trailing `/`
+      page.matchPath = `${page.path}:id`;
+    }
+
+    createPage(page);
+    resolve();
+  });
+};
+```
+
+**Server:**
+
+Links to these pages will work well client-side but fail if a user tries to visit a page directly (i.e. load the page from the server). This is due to your backend server not knowing how to handle the route.
+
+Some server configuration is required to make these types of pages work in production. This configuration will vary depending on your server environment but here's a simple example configuration for NGINX:
+
+```
+location ~ "([a-z]*)\/view\/(\d*)$" {
+    try_files $uri /$1/view/index.html;
+}
+```
+
+This directive will provide the `widgets/view/index.html` file for any request like (widgets)/view/1. It will also support other pages by matching the first level of the URL, example `http://sitename/sprockets/view/1`.
+
+**Client:**
+
+Extracting path params from the route on the client requires that you use the `react-router` `<Route>` in your component. 
```

Comment:
I'm surprised to hear this — the param information should be available without any more work on props.location — is this not so?

Reply:
Ya I was under the impression from convos in discord that you still needed to create client routes for the page. I could not access the route Params without it.

Reply:
Can you verify what the `props.location` looks like w/o the custom client routes? If it's not working, we should fix that in Gatsby's auto-created route components as it's a lot of overhead to make people add their own route components.

Reply:
Will check tomorrow, fairly certain it was empty without the route.

Reply:
Params are not passed to components. 

I searched repository for `matchPath` and it seems gatsby only tests if path is matched and discard return value of react-router `matchPath` (which would have extracted params): https://github.com/gatsbyjs/gatsby/blob/57f7b0d372a7461f019ce86a9003371d45204f3b/packages/gatsby/cache-dir/find-page.js#L35-L47

Reply:
Thanks for the research @pieh — so yeah, we need to fix the problem there — pass into the component the props for the matched path. @gregorskii want to tackle this?

Reply:
Yep I will look into it.

Reply:
@gregorskii btw, do you think you'll still have a chance to look into this?

Reply:
@KyleAMathews I have not had a chance to look into it. I got pulled into another project with a short timeline at work.

I can try to look at it after, but if its a quick fix, does someone else have time?

I was able to solve this on my project with the proposed fix in this original PR, however as discussed my work around may not be needed.